### DISCUSSION TITLE
Problem running with pre-built Docker image

### DISCUSSION THREAD

**Discussion by nshki** [ente-io/ente#1295] [Q&A] [answered]
Hi all. First off, thank you for this lovely project. Having a fully open-source alternative to Google and Apple Photos is amazing and I've been enthusiastic to self-host it on a Raspberry Pi.

I followed [the guide on running using published Docker images here](https://github.com/ente-io/ente/blob/main/server/docs/docker.md) and when I run `docker compose up`, I get all the services to run properly _except_ the `museum` service, which errors out with a:

```
museum-1           | exec ./museum: exec format error
museum-1 exited with code 1
Error response from daemon: cannot join network of a non running container: 06e4d38c54f9d24bb234723c6b668214f17b5f326dfcf508274393b8749685ed
```

Every time I try to run that service. I tried digging a bit, and it seems that the referenced container ID is just referencing the `museum` container, not the other containers.

I then tried going through the guide again without changing any of the config, but I still run into the error. Wanted to see if anyone in the community had an inkling as to what might be a good next move here as I'm out of ideas.

**Comment by mnvr** [comment]
Nothing comes to mind really. I tried the steps again to double check, and they're working for me. So if I had to guess, I'd say somehow your local Docker has gotten into a weird state. 

Things I can suggest:

Delete the existing cluster completely, `docker compose down --volumes --remove-orphans`, delete that directory, restart your Docker daemon (just to make sure there are no random containers running), and then recreate the entire thing in a fresh directory.

Alternatively, you can try doing the setup by building from source instead of using the pre-built images. It sounds like more work, but in fact it is simpler, you just need to clone the repository first

```sh
git clone https://github.com/ente-io/ente
cd ente/server
docker compose up --build
```

**Reply by nshki** [reply, marked as answer]
Thanks for the quick reply. I opted to build from the repo like you suggested and ran into another issue where Docker was failing to pull some dependencies and errored out with `temporary error`. After searching around a bit, I came across [this GitHub comment](https://github.com/gliderlabs/docker-alpine/issues/334#issuecomment-450598069) that did the trick for me.

You were right, my Docker was somehow in a weird state where DNS was failing. By explicitly giving it a DNS address, everything was resolved afterwards.

Things are working well now though! Thanks again for the quick reply.

**Reply by mnvr** [reply]
Awesome!
